testsuite/diem-swarm/src/swarm.rs

```diff
@@ -329,29 +427,28 @@ impl DiemSwarm {
             let node = DiemNode::launch(
                 &self.diem_node_bin_path,
                 node_id.clone(),
-                self.role,
+                self.node_type,
                 &path,
                 logs_dir_path.join(format!("{}.log", index)),
             )
             .unwrap();
             self.nodes.insert(node_id, node);
         }
-        let expected_peers = match self.role {
-            RoleType::Validator => self.nodes.len() - 1,
-            RoleType::FullNode => 1,
-        };
         self.wait_for_startup()?;
+
+        // TODO: Maybe wait for mroe than one on full nodes
```

Comment:
What's the reason we don't wait for them all?

Reply:
We only need 1 connection to guarantee that the network is connected.

It became more complicated to determine "how many" should be connected, especially in a situation where you can only have one directional connections on nodes.  Additionally, I didn't want the tests to take too long, they already take a long time.